bgpd/bgp_vty.c

```diff
@@ -14585,6 +14588,17 @@ static void bgp_show_peer(struct vty *vty, struct peer *p, bool use_json,
 				p->shared_network ? "shared network"
 						  : "non shared network");
 		}
+	} else {
```

Comment:
Isn't this the same as dropping `if (p->su_local) {` and that's it? `%pI6, %pI4` will print (null), not Unknown only.

Reply:
If in_addr or in6_add is NULL, then (null) will be printed.

Reply:
Understood.
But nexthop.v4, nexthop.v6_global and nexthop.v6_local are struct members not a pointer.  
We cannot simply drop if (p->su_local) to reuse the following code, if "null" is the expected values. 
            json_object_string_addf(json_neigh, "nexthop", "%pI4",
                        &p->nexthop.v4);
Instead,  it will make "if (p->su_local)" everywhere like:
if (p->su_local)
        json_object_string_addf(json_neigh, "nexthop", "%pI4",
                        &p->nexthop.v4);
else
       json_object_string_addf(json_neigh, "nexthop", "%pI4", NULL);
       //or json_object_string_addf(json_neigh, "nexthop", "NULL");

However, the struct peer object was calloc(), means bzero'ed.
0.0.0.0 and :: can be printed.
Let me check if it is good enough to represent an "invalid" ip here.

Reply:
I think I actually prefer "unknown" or "not set" to something that will look like a default (?)

Reply:
Then we can change the default to be "Unknown" for print frr extension?

Reply:
@ton31337 -> I really think this is kind of beyond what @karlquan should be doing for this.  There are a *bunch* of places that this would need to be looked at and fixed to get it consistent and also to ensure that we don't just accidently start breaking someones script when the change in display.

Reply:
👌